domains/authorization/Authorization.API/API.UnitTests/API.UnitTests/[Queries]/GetConsentForClientQueryTests.cs

```diff
@@ -11,17 +10,22 @@ public class GetConsentForClientQueryTests
     [Fact]
     public async Task Return_External_Client_When_Everything_Is_Awesome()
     {
-        // Arrange
         var clientRepository = new FakeClientRepository();
         var client = Any.Client();
         await clientRepository.AddAsync(client, CancellationToken.None);
         var sut = new GetConsentForClientQueryHandler(clientRepository);
 
-        // Act
         var result = await sut.Handle(new GetConsentForClientQuery(client.IdpClientId.Value), CancellationToken.None);
 
-        // Assert
-
         result.Should().NotBe(null);
     }
+
+    [Fact]
+    public async Task GivenUnknownClientId_WhenQueryingForAuthorization_ClientIsNotFound()
+    {
+        var clientRepository = new FakeClientRepository();
+        var sut = new GetConsentForClientQueryHandler(clientRepository);
+
+        await Assert.ThrowsAsync<EntityNotFoundException>(async () => await sut.Handle(new GetConsentForClientQuery(Any.Guid()), CancellationToken.None));
```

Comment:
Nit: Can we use fluent assertions?

Reply:
You prefer this:

var action = async () => await sut.Handle(new GetConsentForClientQuery(Any.Guid()), CancellationToken.None);        await action.Should().ThrowAsync<EntityNotFoundException>();

instead of:

await Assert.ThrowsAsync<EntityNotFoundException>(async () => await sut.Handle(new GetConsentForClientQuery(Any.Guid()), CancellationToken.None));

?

Reply:
Maybe this?:

```var clientRepository = new FakeClientRepository();
var sut = new GetConsentForClientQueryHandler(clientRepository);

Func<Task> act = async () => await sut.Handle(new GetConsentForClientQuery(Any.Guid()), CancellationToken.None);

await act.Should().ThrowAsync<EntityNotFoundException>();
```